src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexReadOperation.cs

```diff
@@ -863,9 +865,12 @@ public IEnumerable<string> DynamicEntriesFields(HashSet<string> staticFields)
 
         public override void Dispose()
         {
-            _analyzer?.Dispose();
-            _releaseSearcher?.Dispose();
-            _releaseReadTransaction?.Dispose();
+            using (_readLock)
```

Comment:
This might result in `SynchronizationLockException`. Please note that a query might not be handled by the same thread from the beginning to the end of processing. During the execution we have some `await` calls. So you are not guaranteed that during `Dispose()` will be called on the same thread as `_luceneCleaner.EnterRunningQueryReadLock();`

Reply:
changed